pipeline-model-definition/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/Utils.java

```diff
@@ -22,257 +22,238 @@
+import org.jenkinsci.plugins.workflow.graphanalysis.*;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.jenkinsci.plugins.workflow.job.WorkflowRun;
+import org.jenkinsci.plugins.workflow.job.properties.PipelineTriggersJobProperty;
+import org.jenkinsci.plugins.workflow.steps.FlowInterruptedException;
+import org.jenkinsci.plugins.workflow.steps.Step;
+import org.jenkinsci.plugins.workflow.steps.StepDescriptor;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
+import edu.umd.cs.findbugs.annotations.CheckForNull;
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+import javax.lang.model.SourceVersion;
+
+import java.util.concurrent.TimeUnit;
+import java.util.*;
+import java.util.stream.Collectors;
+
+import groovy.lang.Closure;
+import groovy.lang.MissingPropertyException;
 
 /**
  * Utility methods for use primarily in CPS-transformed code to avoid excessive global whitelisting.
  *
  * @author Andrew Bayer
  */
-@SuppressFBWarnings(value = ["SE_NO_SERIALVERSIONID", "LI_LAZY_INIT_STATIC"])
-class Utils {
+@SuppressFBWarnings(value = {"SE_NO_SERIALVERSIONID", "LI_LAZY_INIT_STATIC"})
+public class Utils {
 
     /**
      * Get the value for this name. First, check the script's properties, then parameters, and finally use the default
      * value, which is set at parse time.
      */
-    static Object getScriptPropOrParam(CpsScript script, String name) {
+    public static Object getScriptPropOrParam(CpsScript script, String name) {
         try {
-            return script.getProperty(name)
+            return script.getProperty(name);
         } catch (MissingPropertyException e) {
-            return script.getProperty('params')?.get(name)
+            return script.getProperty("params") != null ? ((Map) script.getProperty("params")).get(name) : null;
         }
     }
 
     @Restricted(NoExternalUse.class)
-    static Map<String,Closure> getCredsFromResolver(Environment environment, CpsScript script) {
+    public static Map<String, Closure> getCredsFromResolver(Environment environment, CpsScript script) {
         if (environment != null) {
-            environment.credsResolver.setScript(script)
-            return environment.credsResolver.closureMap
+            environment.getCredsResolver().setScript(script);
+            return environment.getCredsResolver().getClosureMap();
         } else {
-            return [:]
+            return new HashMap<>();
         }
     }
 
-    /**
-     * This exists for pre-1.1.2 -> later upgrades of running builds only.
-     *
-     * @param environment The environment to pull credentials from
-     * @return A non-null but possibly empty map of strings to {@link CredentialWrapper}s
-     */
-    @NonNull
-    @Restricted(NoExternalUse.class)
-    static Map<String, CredentialWrapper> getLegacyEnvCredentials(@NonNull Environment environment) {
-        Map<String, CredentialWrapper> m = [:]
-        environment.each {k, v ->
```

Comment:
No longer compilable. Introduced in #147 but (AFAICT) without test coverage, and not clear it ever worked.